include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -376,6 +370,19 @@ class FEEvaluationBase
    */
   gradient_type get_gradient (const unsigned int q_point) const;
 
+  /**
+   * Return the gradient of a finite element function at quadrature point
+   * number @p q_point after a call to @p evaluate(...,true,...) multiplied by
+   * the normal vector on a face. This call is equivalent to calling
+   * `get_gradient() * get_normal_vector` but will use a more efficient
+   * internal representation of data.
+   *
+   * Note that the derived class FEEvaluationAccess overloads this operation
+   * with specializations for the scalar case (n_components == 1) and for the
+   * vector-valued case (n_components == dim).
+   */
+  value_type get_normal_gradient (const unsigned int q_point) const;
```

Comment:
> A more mathematical precise name would be get_normal_derivative, but that somehow hides the fact that we get the gradient and multiply it by the normal vector. Opinions?

I would actually call it `get_normal_derivative` as what you return is a scalar, so it can't be a gradient. I don't think it hides the context of multiplication, especially if you explicitly say that
```
u'(\mathbf x_q) = \boldsymbol \nabla u(\mathbf x_q) \cdot \mathbf n(\mathbf x_q)
```